File: artifacts/api-server/src/lib/lang.ts
// Lightweight language + intent detection. No external LLM required.

const ARABIC_RE = /[\u0600-\u06ff]/;

export function detectLanguage(text: string): "ar" | "en" {
  return ARABIC_RE.test(text) ? "ar" : "en";
}

// English + Arabic triggers for explicit human-handoff requests.
const HUMAN_TRIGGERS_EN = [
  "talk to human",
  "talk to a human",
  "speak to human",
  "speak to a human",
  "human agent",
  "real person",
  "live agent",
  "speak to someone",
  "agent please",
];
const HUMAN_TRIGGERS_AR = [
  "تحدث مع إنسان",
  "موظف حقيقي",
  "موظف بشري",
  "أريد التحدث مع شخص",
  "تحدث مع موظف",
  "وكيل بشري",
  "ممثل خدمة",
];
const COMPLAINT_TRIGGERS_EN = [
  "complain",
  "complaint",
  "terrible",
  "awful",
  "refund",
  "angry",
  "unacceptable",
  "horrible",
  "scam",
];
const COMPLAINT_TRIGGERS_AR = ["شكوى", "غاضب", "سيئ", "استرداد", "غير مقبول"];

export type Intent = "human_request" | "complaint" | "general";

export function classifyIntent(text: string): Intent {
  const lower = text.toLowerCase();
  if (HUMAN_TRIGGERS_EN.some((t) => lower.includes(t))) return "human_request";
  if (HUMAN_TRIGGERS_AR.some((t) => text.includes(t))) return "human_request";
  if (COMPLAINT_TRIGGERS_EN.some((t) => lower.includes(t))) return "complaint";
  if (COMPLAINT_TRIGGERS_AR.some((t) => text.includes(t))) return "complaint";
  return "general";
}

File: artifacts/api-server/app/services/lang.py
import re

_ARABIC_RE = re.compile(r"[؀-ۿ]")

_HUMAN_TRIGGERS_EN = [
    "talk to human", "speak to someone", "live agent", "real person",
    "human agent", "customer service", "speak to agent", "talk to agent",
    "connect me", "escalate", "need help from human",
]
_HUMAN_TRIGGERS_AR = [
    "تحدث مع إنسان", "موظف حقيقي", "وكيل بشري", "خدمة العملاء",
    "تحدث مع موظف", "تواصل مع موظف",
]
_COMPLAINT_TRIGGERS_EN = [
    "complain", "complaint", "refund", "angry", "scam", "fraud",
    "terrible", "horrible", "awful", "disgusting", "rip off", "lawsuit",
]
_COMPLAINT_TRIGGERS_AR = [
    "شكوى", "غاضب", "استرداد", "احتيال", "سيء", "مروع",
]


def detect_language(text: str) -> str:
    return "ar" if _ARABIC_RE.search(text) else "en"


def classify_intent(text: str) -> str:
    lower = text.lower()
    for trigger in _HUMAN_TRIGGERS_EN + _HUMAN_TRIGGERS_AR:
        if trigger in lower:
            return "human_request"
    for trigger in _COMPLAINT_TRIGGERS_EN + _COMPLAINT_TRIGGERS_AR:
        if trigger in lower:
            return "complaint"
    return "general"
